src/bin/mdbook-xgettext.rs

```diff
@@ -81,21 +90,242 @@ fn create_catalog(ctx: &RenderContext) -> anyhow::Result<Catalog> {
+                    path.display().to_string(),
+                    Catalog::new(generate_catalog_metadata(&ctx)),
+                );
+                path.display().to_string()
+            } else {
+                default_file_path.display().to_string()
+            };
+
+            let catalog: &mut Catalog = catalogs
+                .get_mut(&rel_file_path)
+                .with_context(|| anyhow!("Expected to find {:?} in map", rel_file_path))?;
+
             for (lineno, msgid) in extract_messages(&chapter.content) {
                 let source = format!("{}:{}", path.display(), lineno);
-                add_message(&mut catalog, &msgid, &source);
+                add_message(catalog, &msgid, &source);
             }
+
+            let file_path = if depth == 0 {
+                Some(default_file_path.display().to_string())
+            } else if depth == 1 {
+                Some(chapter.path.clone().unwrap().display().to_string())
+            } else {
+                None
+            };
+            get_subcontent_for_chapter(chapter.clone(), file_path, depth)
+                .iter()
+                .for_each(
+                    |ChapterTemplateInfo {
+                         content,
+                         source,
+                         template_destination,
+                     }| {
+                        extract_messages(content)
+                            .iter()
+                            .for_each(|(lineno, msgid)| {
+                                let path = if depth == 0 {
+                                    template_destination.to_owned()
+                                } else {
+                                    ctx.config
+                                        .book
+                                        .src
+                                        .join(template_destination)
+                                        .display()
+                                        .to_string()
+                                };
+                                if let None = catalogs.get(&path) {
+                                    catalogs.insert(
+                                        path.to_string(),
+                                        Catalog::new(generate_catalog_metadata(&ctx)),
+                                    );
+                                }
+                                let catalog = catalogs.get_mut(&path).unwrap();
+
+                                let source2 = format!(
+                                    "{}:{}",
+                                    ctx.config.book.src.join(source).display(),
+                                    lineno
+                                );
+                                add_message(catalog, &msgid, &source2);
+                            });
+                    },
+                );
         }
     }
 
-    Ok(catalog)
+    Ok(catalogs)
+}
+
+// Contains information necessary to create polib messages out
```

Comment:
You're invited to make this a docstring with `///` — even though the type isn't public, it is simpler to just make all documentation comments docstrings.